src/graph.rs

```diff
@@ -528,16 +528,53 @@ impl std::fmt::Display for Value {
 impl Serialize for Value {
     fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
         match self {
-            Value::Null => serializer.serialize_none(),
-            Value::Boolean(value) => serializer.serialize_bool(*value),
-            Value::Integer(value) => serializer.serialize_u32(*value),
-            Value::String(value) => serializer.serialize_str(value),
-            // FIXME: there's no way to distinguish sets and lists, so we can't roundtrip accurately
-            Value::List(list) => serializer.collect_seq(list),
-            Value::Set(set) => serializer.collect_seq(set),
-            // FIXME: we don't distinguish between syntax tree node IDs, graph node IDs, and integers
-            Value::SyntaxNode(node) => serializer.serialize_u32(node.index),
-            Value::GraphNode(node) => serializer.serialize_u32(node.0),
+            Value::Null => {
+                let mut map = serializer.serialize_map(None)?;
+                map.serialize_entry("type", "null")?;
+                map.end()
```

Comment:
The sparse representation for `Value::Null` is unique in not having any kind of value field.